src/voting-metadata-api/src/env.ts

```diff
@@ -0,0 +1,31 @@
+export const DEFAULT_REGION = 'us-west-1'
+export const DEFAULT_CACHE_BUCKET = 'nft-voting-tool-api-developer-cache'
+export const CACHE_BUCKET_NAME = process.env.CACHE_BUCKET_NAME || DEFAULT_CACHE_BUCKET
+export const AWS_REGION = process.env.AWS_REGION || DEFAULT_REGION
+export const IPFS_API_TOKEN = process.env.IPFS_API_TOKEN
+export const isDevelopment = process.env.NODE_ENV === 'development'
+
+/**
+ * Assert that the ENV is in a valid configuration
+ */
+export function assertValidEnv() {
+  if (typeof process.env.ALLOWED_ADDRESSES !== 'string') {
+    throw new Error('Must have valid ALLOWED_ADDRESS list!')
+  }
+  if (typeof process.env.IPFS_API_TOKEN === 'undefined') {
+    throw new Error('Must have valid IPFS_API_TOKEN!')
+  }
+  if (!isDevelopment) {
+    if (typeof process.env.AWS_ACCESS_KEY_ID === 'undefined' || typeof process.env.AWS_ACCESS_KEY_SECRET === 'undefined') {
+      throw new Error('Must have valid AWS_ACCESS_KEY_ID and AWS_ACCESS_KEY_SECRET')
+    }
```

Comment:
This wouldn't normally be defined at runtime (I think, although it's possible AWS injects them) - we pass them into the cdk deployment, but that's deploy time, not run time.

Reply:
Should be injected and available to the application, was thinking it would be needed for the S3 upload functionality. Will double check

Reply:
Looks like they have some magic baked into `@aws/*` with their `config` interface. Should be available in every package from what I can tell from the docs.

This does the trick and should work in any valid configuration 🕺:
```javascript
await s3.config.credentials()
```